datasetIngestor/sendIngestCommand_test.go

```diff
@@ -0,0 +1,231 @@
+	}
+}
+
+func TestGetEndpoint(t *testing.T) {
+	// Redirect log output to a buffer
+	var buf bytes.Buffer
+	log.SetOutput(&buf)
+	defer func() {
+		log.SetOutput(os.Stderr)
+	}()
+
+	testCases := []struct {
+		dsType string
+		want   string
+	}{
+		{"raw", "/RawDatasets"},
+		{"derived", "/DerivedDatasets"},
+		{"base", "/Datasets"},
+		{"unknown", ""},
+	}
+
+	for _, tc := range testCases {
+		got, err := getEndpoint(tc.dsType)
+		if err != nil && tc.dsType != "unknown" {
+			t.Errorf("getEndpoint(%q) returned unexpected error: %v", tc.dsType, err)
+		}
+		if got != tc.want {
+			t.Errorf("getEndpoint(%q) = %q; want %q", tc.dsType, got, tc.want)
+		}
+		if tc.dsType == "unknown" && err == nil {
+			t.Errorf("Expected error for unknown dataset type not found")
+		}
+		buf.Reset()
+	}
+}
+
+func TestSendRequest(t *testing.T) {
+	// Create a test server
+	ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		w.WriteHeader(http.StatusOK)
+	}))
+	defer ts.Close()
+	
+	// Create a test client
+	client := &http.Client{}
+	
+	// Call the sendRequest function
+	resp := sendRequest(client, "GET", ts.URL, nil)
+	
+	// Check the response
+	if resp.StatusCode != http.StatusOK {
+		t.Errorf("sendRequest() returned status %d; want %d", resp.StatusCode, http.StatusOK)
+	}
+}
+
+func TestDecodePid(t *testing.T) {
+	// Create a test response
+	resp := &http.Response{
+		Body: io.NopCloser(strings.NewReader(`{"pid": "12345"}`)),
+	}
+	
+	// Call the decodePid function
+	pid := decodePid(resp)
+	
+	// Check the returned pid
+	if pid != "12345" {
+		t.Errorf("decodePid() returned pid %q; want %q", pid, "12345")
+	}
+}
+
+func TestCreateOrigDatablocks(t *testing.T) {
+	// keep track of the number of requests
+	var requestCount int
+	
+	// Create a mock HTTP server
+	server := httptest.NewServer(http.HandlerFunc(func(rw http.ResponseWriter, req *http.Request) {
+		// Increment the request count for each request
+		requestCount++
+		
+		// Check if the request method is POST
+		if req.Method != http.MethodPost {
+			t.Errorf("Expected POST request, got %s", req.Method)
+		}
+		
+		// Check if the request URL is correct
+		expectedURL := "/OrigDatablocks?access_token=testToken"
+		if req.URL.String() != expectedURL {
+			t.Errorf("Expected request to %s, got %s", expectedURL, req.URL.String())
+		}
+		
+		rw.WriteHeader(http.StatusOK)
+	}))
+	// Close the server when test finishes
+	defer server.Close()
+	
+	// Create a mock HTTP client
+	client := server.Client()
+	
+	// Define test data
+	datafiles := []Datafile{
+		{
+			Size: 100,
+		},
+		{
+			Size: 200,
+		},
+		{
+			Size: 900,
+		},
+	}
+	user := map[string]string{
+		"accessToken": "testToken",
+	}
+	
+	// Call the function with test data
+	createOrigDatablocks(client, server.URL, datafiles, "testDatasetId", user)
+	
+	// Check if the correct number of requests were made
+	expectedRequestCount := (len(datafiles) + BLOCK_MAXFILES - 1) / BLOCK_MAXFILES
```

Comment:
isn't `expectedRequestCount` equal to 1? It would be nice to have test cases for different values of expectedRequestCount (so changing BLOCK_MAXFILES) or changing the datafiles len. (if not too much work)

Reply:
It is added